src/compute-client/src/controller.rs

```diff
@@ -705,12 +715,12 @@ impl<T: Timestamp> ComputeInstanceRef<'_, T> {
 
     /// Return a read-only handle to the indicated collection.
     pub fn collection(&self, id: GlobalId) -> Result<&CollectionState<T>, CollectionMissing> {
-        self.instance.collection(id)
+        self.instance.live_collection(id)
     }
 
     /// Return an iterator over the installed collections.
     pub fn collections(&self) -> impl Iterator<Item = (&GlobalId, &CollectionState<T>)> {
-        self.instance.collections_iter()
+        self.instance.live_collections_iter()
```

Comment:
Same question: How can we be sure that a caller won't panic when not finding some collection. But here, it seems easier, since it seems these snapshots are only used for index lookups, where I don't see any panics due to not finding collections; it just won't report any indexes for an id that it couldn't look up.